### PR TITLE
feat: Added PermissionManager and SourceHandler

### PR DESCRIPTION
**Pull request by hasrat-ali**

### CODE REVIEW

File: app/src/main/java/a/a/a/Jx.java [hunk 1]
```diff
@@ -299,20 +315,21 @@ public String a() {
...
         }
-        if (!this.f.a(this.c.getActivityName()).a() || this.c.fileName.contains("_fragment")) {
-            sb.append("initializeLogic();");
-            sb.append(a);
-        } else {
+        sb.append(a);
+        if (!this.c.fileName.contains("_fragment")) {
             jq jqVar = this.f;
-            sb.append(Lx.a(jqVar.g, jqVar.a(this.c.getActivityName()).c));
+            //sb.append(Lx.a(jqVar.g, jqVar.a(this.c.getActivityName()).c));
```
Comment (JavkhlanK) [inline app/src/main/java/a/a/a/Jx.java:321]: Just a hint: you don't have to comment out old code, git will automatically save what was on that line before you made a commit.

Comment (JavkhlanK) [inline app/src/main/java/a/a/a/Jx.java:321]: Just a note: You don't have to comment out lines, git will automatically save what on that line was.

File: app/src/main/java/mod/w3wide/blocks/ExtraBlocks.java [hunk 2]
```diff
@@ -18,6 +19,7 @@
     private final String eventName;
     private final String javaName;
     private final String sc_id;
+    private final String xmlName;
```
Comment (JavkhlanK) [inline app/src/main/java/mod/w3wide/blocks/ExtraBlocks.java:22]: Is this really used anywhere?

File: app/src/main/java/mod/w3wide/blocks/ExtraBlocks.java [hunk 3]
```diff
@@ -57,6 +60,9 @@ public static void extraBlocks(ArrayList<HashMap<String, Object>> arrayList) {
         arrayList.add(0, addBlock("fileGetParent", "s", "", "%s.getParent()", "#a1887f", "%m.File getParent"));
         arrayList.add(0, addBlock("fileGetPath", "s", "", "%s.getPath()", "#a1887f", "%m.File getPath"));
         arrayList.add(0, addBlock("fileIsHidden", "b", "", "%s.isHidden()", "#a1887f", "%m.File isHidden"));
+        //Basically Command Block
+        arrayList.add(0, addBlock("addPermission", " ", "", "", "#493F5A", "Permission Command Block : add %m.Permission"));
+        arrayList.add(0, addBlock("removePermission", " ", "", "", "#493F5A", "Permission Command Block : remove %m.Permission"));
```
Comment (JavkhlanK) [inline app/src/main/java/mod/w3wide/blocks/ExtraBlocks.java:65]: I think these should be next to regular Command Blocks.

File: app/src/main/java/dev/aldi/sayuti/block/ExtraPaletteBlock.java [hunk 4]
```diff
@@ -1103,6 +1093,9 @@ public void setBlock(int i, int i2) {
                     logicEditor.a("Command Blocks", 0xff555555);
                     logicEditor.a("c", "CommandBlockJava");
                     logicEditor.a("c", "CommandBlockXML");
+                    logicEditor.a("Permission Command Blocks", 0xff555555);
+                    logicEditor.a(" ", "addPermission");
+                    logicEditor.a(" ", "removePermission");
```
Comment (JavkhlanK) [inline app/src/main/java/dev/aldi/sayuti/block/ExtraPaletteBlock.java:1098]: Oh, they are already, lol. Forget about my last comment :sweat_smile:

File: app/src/main/java/dev/aldi/sayuti/block/ExtraPaletteBlock.java [hunk 5]
```diff
@@ -84,19 +83,10 @@ private boolean isWidgetUsed(String str) {
...
-            } else {
-                logicEditor.g(ss);
-            }
-        }
-    }
-
+    /**
+    * ExtraPaletteBlock#f(Ss) moved to mod.w3wide.menu.ExtraMenuBean#defineMenuSelector(Ss)
+    * for better block menu selections and to add new stuff easily.
+    */
```
Comment (JavkhlanK) [inline app/src/main/java/dev/aldi/sayuti/block/ExtraPaletteBlock.java:89]: Why is this a Javadoc of `ExtraPaletteBlock#e` though?

File: app/src/main/java/a/a/a/Jx.java (hunk 1 above)
Comment (hasrat-ali) [reply app/src/main/java/a/a/a/Jx.java:321]: I commented out when I am testing and I forgot about that

File: app/src/main/java/dev/aldi/sayuti/block/ExtraPaletteBlock.java (hunk 5 above)
Comment (hasrat-ali) [reply app/src/main/java/dev/aldi/sayuti/block/ExtraPaletteBlock.java:89]: Actually, this javadoc is not for `ExtraPaletteBlock#e`, it's for `ExtraPaletteBlock#f`.

File: app/src/main/java/mod/w3wide/blocks/ExtraBlocks.java (hunk 2 above)
Comment (hasrat-ali) [reply app/src/main/java/mod/w3wide/blocks/ExtraBlocks.java:22]: Ohh, I forgot to remove.

File: app/src/main/java/a/a/a/Jx.java (hunk 1 above)
Comment (hasrat-ali) [reply app/src/main/java/a/a/a/Jx.java:321]: Same, I forgot to remove

**Pull request merged**